File: JavaPractice/src/CountDuplicateNumbers.java
import java.util.*;
import java.util.HashMap;
/**
 * Implement java code for below questions -> Count Duplicate ElementsGiven
 * an integer array, numbers, count the number of elements that occur more than once.
 * Examplenumbers = [1, 3, 3, 4, 4, 4] There are two non-unique elements: 3 and 4.
 * Function DescriptionComplete the function countDuplicate in the editor below.
 * countDuplicate has the following parameter(s): int numbers[n]: an array of integers Returns: int: an
 * integer that denotes the number of non-unique values in the numbers array Constraints3 ≤ n ≤ 10001 ≤ numbers[i] ≤ 1000, 0 ≤ i < n
 * Input Format Format for Custom TestingSample Case 0Sample InputSTDIN Function----- -----8 → numbers[] size n = 8 1 → numbers = [1, 3, 1, 4, 5, 6, 3, 2]3145632
 * Sample Output2
 * @author divyesh.shah
 */
public class CountDuplicateNumbers {
    public static int countDuplicate(int[] numbers) {
        int count = 0;
        Map<Integer, Integer> map = new HashMap<>();

        // Count the frequency of each element in the array
        for (int i = 0; i < numbers.length; i++) {
            int frequency = map.getOrDefault(numbers[i], 0);
            map.put(numbers[i], frequency + 1);
        }

        // Count the number of elements that occur more than once
        for (Map.Entry<Integer, Integer> entry : map.entrySet()) {
            if (entry.getValue() > 1) {
                count++;
            }
        }

        return count;
    }

    public static void main(String[] args) {
        System.out.println(countDuplicate(new int[]{1, 2, 3, 4, 5, 2, 3, 5}));
    }
}

File: PythonPractice/src/CountDuplicateNumbers.py
from typing import List

def count_duplicate(numbers: List[int]) -> int:
    if not isinstance(numbers, list):
        raise ValueError("Input must be a list")

    if any(not isinstance(num, int) for num in numbers):
        raise ValueError("All elements in the list must be integers")

    if len(numbers) < 3 or len(numbers) > 10000:
        raise ValueError("The list length must be between 3 and 10,000")

    if any(num < 1 or num > 1000 for num in numbers):
        raise ValueError("All integers must be between 1 and 1000")

    count = 0
    frequency_map = {}

    # Count the frequency of each element in the list
    for num in numbers:
        if num in frequency_map:
            frequency_map[num] += 1
        else:
            frequency_map[num] = 1

    # Count the number of elements that occur more than once
    for frequency in frequency_map.values():
        if frequency > 1:
            count += 1

    return count

if __name__ == "__main__":
    try:
        # Test case
        print(count_duplicate([1, 2, 3, 4, 5, 2, 3, 5]))  # Output should be 3
    except ValueError as e:
        print(f"Error: {e}")
